### PR TITLE
Minor changes driven by implementing nested locations in WebSkim. Add…

### PR DESCRIPTION
**Pull request by michaelcfanning**
… directory mime-type. Correct pathFromParent computation (all children are prefixed by / character).

@lgolding @rtaket

### CODE REVIEW

File: src/Sarif/Core/FileData.cs [hunk 1]
```diff
@@ -36,13 +36,13 @@ public static IList<FileData> Create(IEnumerable<Uri> uris, out string fileDataK
...
                 {
-                    fileData.PathFromParent = "#" + uri.ToString();
-                    fileDataKey = fileDataKey + fileData.PathFromParent;
+                    fileData.PathFromParent = uri.ToString();
+                    fileDataKey = fileDataKey + "#" + fileData.PathFromParent;
                 }
                 else
                 {
                     Debug.Assert(!uri.IsAbsoluteUri);                    
-                    fileData.PathFromParent = "/" + uri.ToString();
```
Comment (michaelcfanning) [inline src/Sarif/Core/FileData.cs]: path was already relative here, so this added an unnecessary forward slash

File: src/Sarif/Core/FileData.cs [hunk 2]
```diff
@@ -36,13 +36,13 @@ public static IList<FileData> Create(IEnumerable<Uri> uris, out string fileDataK
                 }
                 else if (files.Count == 1)
                 {
-                    fileData.PathFromParent = "#" + uri.ToString();
-                    fileDataKey = fileDataKey + fileData.PathFromParent;
+                    fileData.PathFromParent = uri.ToString();
+                    fileDataKey = fileDataKey + "#" + fileData.PathFromParent;
```
Comment (michaelcfanning) [inline src/Sarif/Core/FileData.cs]: # character is part of the key, but not the pathFromParent data.

File: src/Sarif/Writers/MimeType.cs [hunk 3]
```diff
@@ -10,15 +10,20 @@ namespace Microsoft.CodeAnalysis.Sarif.Writers
     public static class MimeType
     {
         /// <summary>Guesses an appropriate MIME type given the extension from a file name.</summary>
-        /// <param name="fileName">File name from which MIME type shall be guessed.</param>
-        /// <returns>A string corresponding to the likely MIME type of <paramref name="fileName"/> given        public static string DetermineFromFileExtension(Uri fileUri)
+        /// <param name="fileName">File path from which MIME type shall be guessed.</param>
+        /// <returns>A string corresponding to the likely MIME type of <paramref name="filePath"/>
```
Comment (lgolding) [inline src/Sarif/Writers/MimeType.cs]: You changed the paramRef but not the name of the parameter itself.

**Comment by lgolding** [comment]
:+1 with one correction.

File: src/Sarif/Core/FileData.cs (hunk 2 above)
Comment (michaelcfanning) [inline src/Sarif/Core/FileData.cs]: what the heck with the font size here?? :)

Comment (lgolding) [inline src/Sarif/Core/FileData.cs]: You wrote "# characters...", and Markdown interpreted the "#" to mean "first-level heading".

File: src/Sarif/Writers/MimeType.cs (hunk 3 above)
Comment (michaelcfanning) [inline src/Sarif/Writers/MimeType.cs]: fixed

**Pull request merged**